### ISSUE TITLE
Idea: Improved nunjacks "interface"?

### ISSUE THREAD

**Issue by olearydj** [mgmeyers/obsidian-zotero-integration#244]
It would be great if... we could format nunjacks code in a way that makes the structure more obvious without affecting the spacing of the output (e.g. indentation). I know that this idea runs counter to the basic implementation, but perhaps there is a way to get the best of both worlds.

Also, it would be nice if we could put nunjacks code into a code block to prevent styling and use a monospace font.

**Comment by cs-hall** [comment]
I recently found that you can use dashes to remove whitespace preceding or proceeding the output. See the documentation here: https://mozilla.github.io/nunjucks/templating.html#whitespace-control. By doing this most places you can indent the code or add linespaces to make it more readable.

Overall, the [nunjucks documentation](https://mozilla.github.io/nunjucks/templating.html) is pretty good and was well worth the read for me!

**Comment by olearydj** [comment]
@cs-hall yes, that is true. I just have a hard time wrapping my head around it. the nunjucks library and docs are good enough that I find myself spending most time figuring out how to deal with spacing issues
